packages/react-avatar/src/components/Avatar/useAvatar.tsx

```diff
@@ -34,26 +48,40 @@ export const useAvatar_unstable = (props: AvatarProps, ref: React.Ref<HTMLElemen
       defaultProps: {
         children: <PersonRegular />,
         'aria-hidden': true,
+        id: baseId + '__initials',
       },
     });
   }
 
-  // The image's alt text should be the name, but if name is missing, fall back to the initials
-  let alt = name;
-  if (!alt && typeof initials?.children === 'string') {
-    alt = initials.children;
-  }
+  const image: AvatarState['image'] = resolveShorthand(props.image, {
+    defaultProps: {
+      alt: '',
+      role: 'presentation',
+      'aria-hidden': true,
+    },
+  });
+
+  const badge: AvatarState['badge'] = resolveShorthand(props.badge, {
+    defaultProps: {
+      size: getBadgeSize(size),
+      role: 'presentation',
+      'aria-hidden': true,
+      id: baseId + '__badge',
+    },
+  });
 
-  // Resolve the image slot
-  const image = resolveShorthand(props.image, { defaultProps: { alt } });
+  // Resolve aria-label and/or aria-labelledby if not provided by the user
+  if (!root['aria-label'] && !root['aria-labelledby']) {
+    if (name) {
+      root['aria-label'] = name;
 
-  // If there's no image, make either the initials or icon have role="img" and aria-label={alt}
-  if (!image) {
-    const fallbackSlot = initials || icon;
-    if (fallbackSlot) {
-      fallbackSlot.role ??= 'img';
-      fallbackSlot['aria-label'] ??= alt;
-      delete fallbackSlot['aria-hidden'];
+      // Include the badge in labelledby if it exists
+      if (badge) {
```

Comment:
Should this be an `else if` since you shouldn't have both `aria-label` and `aria-labelledby` set at the same time?

Reply:
I think it's okay to have both:

> The purpose of aria-labelledby is the same as that of aria-label. It provides the user with a recognizable, accessible name for an interactive element. If an element has both attributes set, aria-labelledby will be used. aria-labelledby takes precedence over all other methods of providing an accessible name, including aria-label, label, and the element's inner text.

[Source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)

That said, since this is something we're setting and not the user of the component it might be nice to apply only one.

Reply:
It's actually deliberately setting both. The aria-label is the user's name, and aria-labelledby is the avatar itself, plus the badge's label. It's effectively concatenating the name and badge status.

The result is a tree like this, and the final label being "Ben Howell busy"
```html
<span id="avatar-42" aria-label="Ben Howell" aria-labelledby="avatar-42  avatar-42__badge">
   <!-- ... -->
   <span id="avatar-42__badge" aria-label="busy" />
</span>
```